### PR TITLE
Add support for JSON in FlowFuse Assistant

### LINKED ISSUE #4126: Adjust assistant API Route for JSON prompts
```[tasklist]
### Tasks
- [x] Update assistant route
- [x] Update docs to refect support for JSON editors
- [x] Update tests
```

### PR DESCRIPTION
**Pull request by Steve-Mcl**
closes #4126

## Description

Adds support for JSON assistance

## Related Issue(s)

#4126

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [x] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: test/unit/forge/routes/api/assistant_spec.js [hunk 1]
```diff
@@ -92,73 +92,112 @@ describe('Assistant API', async function () {
...
-                headers: { authorization: 'Bearer ' + TestObjects.tokens.instance },
-                payload: { prompt: 'multiply by 5' }
+    })
+
+    describe('service tests', async function () {
+        function serviceTests (serviceName) {
+            it('anonymous cannot access', async function () {
+                const response = await app.inject({
+                    method: 'GET',
+                    url: `/api/v1/assistant/${serviceName}n`
```
Comment (knolleary) [inline test/unit/forge/routes/api/assistant_spec.js:131]:
```suggestion
                    url: `/api/v1/assistant/${serviceName}`
```

Typo?

File: forge/routes/api/assistant.js [hunk 2]
```diff
@@ -45,8 +45,10 @@ module.exports = async function (app) {
         }
     },
     async (request, reply) => {
-        // const method = request.params.method // FUTURE: allow for different methods
-        const method = 'function' // for now, only function node/code generation is supported
+        const method = request.params.method // the method to call at the assistant service
+        if (/^[a-z0-9_-]+$/.test(method) === false) {
```
Comment (knolleary) [inline forge/routes/api/assistant.js:49]: Should we test for the allowed functions here - ie `function`/`json`?

Comment (Steve-Mcl) [reply forge/routes/api/assistant.js:49]: I did consider that but I can (and do) evaluate them at the serice end.

File: test/unit/forge/routes/api/assistant_spec.js (hunk 1 above)
Comment (Steve-Mcl) [reply test/unit/forge/routes/api/assistant_spec.js:131]: yeah, good spot. FYI, it passes because it is a valid param name.

**Review by knolleary — APPROVED** [review]

**Pull request merged**